mono/utils/mono-merp.c

```diff
@@ -0,0 +1,500 @@
+/**
+ * \file
+ * Support for interop with the Microsoft Error Reporting tool
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+
+#include <config.h>
+#include <glib.h>
+
+#ifdef HOST_DARWIN
+#include "mono-merp.h"
+
+#include <unistd.h>
+#include <spawn.h>
+
+// OSX OS stuff now, for merpGUI interop
+#include <mach/mach.h>
+#include <mach/task_info.h>
+#include <mach/mach_types.h>
+#include <mach/mach_traps.h>
+#include <servers/bootstrap.h>
+
+typedef enum {
+	MerpArchInvalid = 0,
+
+	MerpArchx86_64 = 1,
+	MerpArchx86 = 2,
+	MerpArchPPC = 3,
+	MerpArchPPC64 = 4
+} MerpArch;
+
+typedef enum
+{
+	MERP_EXC_NONE = 0,
+
+	MERP_EXC_FORCE_QUIT = 1,
+	MERP_EXC_SIGSEGV = 2,
+	MERP_EXC_SIGABRT = 3,
+	MERP_EXC_SIGSYS  = 4,
+	MERP_EXC_SIGILL = 5,
+	MERP_EXC_SIGBUS = 6,
+	MERP_EXC_SIGFPE = 7 ,
+	MERP_EXC_SIGTRAP = 8,
+	MERP_EXC_SIGKILL = 9,
+	MERP_EXC_HANG  = 10
+} MERPExcType;
+
+typedef struct {
+	pid_t pidArg; // Process ID of crashed app (required for crash log generation)
+	MerpArch archArg; // Arch, MacOS only, bails out if not found also required for bucketization. (required)
+	uintptr_t capabilitiesArg; // App capabilities (optional) i.e. recover files, etc.
```

Comment:
You could use `gsize` and `gchar` instead of `uintptr_t`/`size_t` and `char` respectively.